### PR TITLE
Improve Prompt & Introduce Cluster Status Banner Message

### PR DESCRIPTION
**Pull request by ColonelPanicks**
This PR makes a few changes: 

- Adds options to 'set' for toggling active banner parts (currently for the 'tips' and 'status' sections)
    - Adds corresponding information to 'info'
- Introduces 'status' banner part
    - This is useful for Flight Solo (and any other systems utilising Flight Profile & Flight Hunter)
    - Disabled by default
- Adds flight set to the active banner command tips
- Expands flight set help page to give similar info to flight info on the different keys that can be set (may be slightly redundant but I did it before realising info gives some of this info (and more), I think it's worth keeping though) 

An example of what the prompt can look like when cluster status is enabled 
Screenshot 2024-01-16 at 17 52 02

### CODE REVIEW

File: dist/opt/flight/etc/banner/banner.d/10-tips.sh [hunk 1]
```diff
@@ -25,6 +25,9 @@
 # https://github.com/openflighthpc/flight-starter
 #==============================================================================
 (
+  if [[ ${flight_STARTER_tips:-enabled} != "enabled" ]] ; then
+    exit
```
Comment (mjtko) [inline dist/opt/flight/etc/banner/banner.d/10-tips.sh:29]: I think this should be a `return` rather than an `exit` as these files are `source`d rather than executed -- so a sourced `exit` will exit the sourcer, which I don't think is intended, and would prevent `20-status.sh` from running, for example.

Might be wrong though!

File: dist/opt/flight/etc/banner/banner.d/20-status.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  # Don't show unless enabled with 'flight set status on'
+  if [[ ${flight_STARTER_status:-'disabled'} != "enabled" ]] ; then
+    exit
+  fi
+
+  # Skip if flight-profile and flight-hunter are not present
+  if ! [[ -f ${flight_ROOT}/libexec/commands/profile && -f ${flight_ROOT}/libexec/commands/hunter ]] ; then
+    echo "Cluster Status: Not showing (Profile & Hunter not found)"
+    exit
```
Comment (mjtko) [inline dist/opt/flight/etc/banner/banner.d/20-status.sh:41]: The `exit` calls in this file should probs also be `return` calls.

File: dist/opt/flight/etc/banner/tips.d/02-set.rc [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#==============================================================================
+# Copyright (C) 2019-present Alces Flight Ltd.
```
Comment (mjtko) [inline dist/opt/flight/etc/banner/tips.d/02-set.rc:2]: 2024? :)

File: dist/opt/flight/etc/banner/banner.d/20-status.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
+#==============================================================================
+# Copyright (C) 2019-present Alces Flight Ltd.
```
Comment (mjtko) [inline dist/opt/flight/etc/banner/banner.d/20-status.sh:2]: 2024? :)

File: dist/opt/flight/libexec/commands/set [hunk 5]
```diff
@@ -133,9 +133,25 @@ case $action in
...
+      welcome       Toggle showing of welcome to cluster line (when 'always' is off)
+
+      secondary     Toggle running of ${flight_STARTER_product} on secondary shells
+
+      always        Set whether ${flight_STARTER_product} is activated on login 
+
+      tips          Set ${flight_STARTER_product} sub-command tips (when 'always' is on)
+
+      status        Set showing of ${flight_STARTER_product} cluster status
+                    (when 'always' is on)
```
Comment (mjtko) [inline dist/opt/flight/libexec/commands/set:153]: Why is this only active when `always` is on? Doesn't the status (and tips) appear when `flight start` is executed if `always` isn't on?

**Review by mjtko — CHANGES_REQUESTED** [review]
This implementation is fine, just a general comment:

Usually we'd bundle tool-specific enhancements for other `flight` tools (including `flight-starter`) with those tools, i.e. `etc/banner/banner.d/20-status.sh` would be installed as part of `profile`/`hunter` -- or both, somehow? can they exist separately from each other?

Looking at it, possibly the `20-status.sh` could maybe be split up for `profile` and `hunter`.

Obvs, pulling this banner enhancement out to the specific packages would also indicate that the associated `set` enhancements should also be pulled out there, but I don't think there's a plugin mechanism in `set` right now. That's a possible future enhancement and would be a more general solution and opens up adding further set-based tweaks in future.

Thinking aloud: another approach would be the one taken with `flight env` which installs its own `profile.d` script for setting the default environment, which it manages itself. `flight hunter` and `flight profile` could take this approach rather than hooking into the banner.

TL;DR: looks like this works (though I think the inline comments should be addressed), and there are possible opportunities for refactoring/enhancements that exist. :)

File: dist/opt/flight/etc/banner/banner.d/10-tips.sh (hunk 1 above)
Comment (ColonelPanicks) [reply dist/opt/flight/etc/banner/banner.d/10-tips.sh:29]: Seems to work for me but if it's better practice then I shall adapt it.

Evidence: 
```
stu@furugoori ~$ ssh 10.151.15.211                                                                                                                                    255 ↵
Warning: Permanently added '10.151.15.211' (ED25519) to the list of known hosts.
Last login: Wed Jan 17 10:18:44 2024 from 10.151.0.54
  'o`
 'ooo`                                Welcome to your cluster
 `oooo`
  `oooo`         'o`                       Flight Solo 2023.6
    `ooooo`  `ooooo                  Based on Rocky Linux 9.2
       `oooo:oooo`
          `v  -[ alces flight ]-

TIPS:

'flight help'           - get help on available commands
'flight set'            - change login defaults (see 'flight info' for details)
'flight env'            - manage software package ecosystems
'flight desktop'        - manage interactive GUI desktop sessions

CLUSTER STATUS:

Hunted Nodes
  Parsed:    1    Buffer:    2

Configured Nodes
  Completed:    1    Applying:    1    Failed:    0

[flight@prompt-experiment1 ~]$ flight set tips off
flight set: Flight Solo prompt tips disabled.
[flight@prompt-experiment1 ~]$
logout
Connection to 10.151.15.211 closed.
stu@furugoori ~$ ssh 10.151.15.211
Warning: Permanently added '10.151.15.211' (ED25519) to the list of known hosts.
Last login: Wed Jan 17 10:32:19 2024 from 10.151.0.1
  'o`
 'ooo`                                Welcome to your cluster
 `oooo`
  `oooo`         'o`                       Flight Solo 2023.6
    `ooooo`  `ooooo                  Based on Rocky Linux 9.2
       `oooo:oooo`
          `v  -[ alces flight ]-

CLUSTER STATUS:

Hunted Nodes
  Parsed:    1    Buffer:    2

Configured Nodes
  Completed:    1    Applying:    1    Failed:    0

[flight@prompt-experiment1 ~]$
```

File: dist/opt/flight/etc/banner/tips.d/02-set.rc (hunk 3 above)
Comment (ColonelPanicks) [reply dist/opt/flight/etc/banner/tips.d/02-set.rc:2]: Because new files? Yep, makes sense!

File: dist/opt/flight/libexec/commands/set (hunk 5 above)
Comment (ColonelPanicks) [reply dist/opt/flight/libexec/commands/set:153]: Good point, I will adapt to mention for "active flight environment" or something to imply when `flight start` is run or `always on` is set

**Comment by ColonelPanicks** [comment]
Closing in favour of #17 due to implementing things more "properly" (e.g. tools add their own tips, set commands and banner files)
